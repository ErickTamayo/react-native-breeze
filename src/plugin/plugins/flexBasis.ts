import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^flex-basis-(?<key>${keys("flexBasis")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["flexBasis", key]);
  return { flexBasis: value };
};
