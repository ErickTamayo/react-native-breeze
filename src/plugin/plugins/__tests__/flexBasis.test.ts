import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../flexBasis";

describe("flexBasis", () => {
  // prettier-ignore
  const shouldMatch = generateInput("flex-basis", ["1", "2", "3", "4", "5", "6", "8"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "flex-basis-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
