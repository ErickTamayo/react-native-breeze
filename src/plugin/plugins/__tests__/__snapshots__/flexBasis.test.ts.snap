// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexBasis Should parse (flex-basis-1) correctly 1`] = `
Object {
  "flexBasis": 4,
}
`;

exports[`flexBasis Should parse (flex-basis-2) correctly 1`] = `
Object {
  "flexBasis": 8,
}
`;

exports[`flexBasis Should parse (flex-basis-3) correctly 1`] = `
Object {
  "flexBasis": 12,
}
`;

exports[`flexBasis Should parse (flex-basis-4) correctly 1`] = `
Object {
  "flexBasis": 16,
}
`;

exports[`flexBasis Should parse (flex-basis-5) correctly 1`] = `
Object {
  "flexBasis": 20,
}
`;

exports[`flexBasis Should parse (flex-basis-6) correctly 1`] = `
Object {
  "flexBasis": 24,
}
`;

exports[`flexBasis Should parse (flex-basis-8) correctly 1`] = `
Object {
  "flexBasis": 32,
}
`;
